src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/IssueService.java

```diff
@@ -36,13 +48,42 @@
     public static final String API_PATH_TRANSITIONS_SUFFIX = "transitions";
     public static final String API_PATH_COMMENTS_SUFFIX = "comment";
 
-    private JiraCloudService jiraCloudService;
+    private final JiraCloudService jiraCloudService;
+    private final UserSearchService userSearchService;
+    private final ProjectService projectService;
+    private final IssueTypeService issueTypeService;
 
-    public IssueService(final JiraCloudService jiraCloudService) {
+    public IssueService(final JiraCloudService jiraCloudService, final UserSearchService userSearchService, final ProjectService projectService, final IssueTypeService issueTypeService) {
         this.jiraCloudService = jiraCloudService;
+        this.userSearchService = userSearchService;
+        this.projectService = projectService;
+        this.issueTypeService = issueTypeService;
+    }
+
+    public IssueResponseModel createIssue(IssueCreationRequestModel requestModel) throws IntegrationException {
+        IssueRequestModelFieldsBuilder fieldsBuilder = new IssueRequestModelFieldsBuilder();
+        fieldsBuilder.copyFields(requestModel.getFieldsBuilder());
+        Optional<IssueTypeResponseModel> bugIssueType = issueTypeService.getAllIssueTypes().stream()
```

Comment:
Since I think this is now variable, we might want to change it to just issueType